servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancerFactory.java

```diff
@@ -130,17 +126,23 @@ public Builder() {
          * Configures the {@link RoundRobinLoadBalancerFactory} to produce a {@link LoadBalancer} with
          * a setting driving eagerness of connection shutdown. When configured with {@code false} as the argument,
          * the created {@link LoadBalancer} does not close connections when a host becomes
-         * {@link ServiceDiscovererEvent#isAvailable() unavailable}. If the value is {@code true},
+         * {@link ServiceDiscovererEvent.Status#UNAVAILABLE unavailable}. If the value is {@code true},
          * the connections will be closed gracefully on such event.
          *
          * @param eagerConnectionShutdown when {@code true}, connections will be shut down upon receiving
-         * {@link ServiceDiscovererEvent#isAvailable() unavailable} events for a particular host. Value of {@code false}
-         * preserves connections and routes requests through them but no new connections are opened for such host.
+         * {@link ServiceDiscovererEvent.Status#UNAVAILABLE unavailable} {@link ServiceDiscovererEvent events}
+         * for a particular host. Value of {@code false} preserves connections and routes requests through them
+         * but no new connections are opened for such host.
          * @return {@code this}.
+         * @deprecated This configuration has no effect. To control the behaviour, configure the
```

Comment:
Thanks for describing in this comment how users should migrate. However, when we deprecate something, we should not kill functionality. With the new SD contract, we can still make this feature work as expected: apply a `map` operator for SD publisher and depending on `eagerConnectionShutdown` flag, flip `EXPIRED` -> `UNAVAILABLE` or `UNAVAILABLE` -> `EXPIRED`.

Reply:
Agreed, killing the feature was a mistake. I brought it back. However, there's one drawback with it. The users won't be able to handle three states separately. If LB was able to distinguish expired from unavailable and provide unavailable events as well as expired, it will not be possible to turn off the mapping. Now users have to coordinate between the state the LB emits (in case of the built-in implementation it's DNS) and the flag here. It's impossible to swap the new default `EXPIRED` to `UNAVAILABLE` and expect everything to work if the previous setting of eagerConnectionShutdown was the default `false`.  The new state was just introduced, perhaps it's worth considering a non-binary setting here -> "If not set, accept the raw state and act upon it".

Reply:
Good point! 🎖️ 
I like suggestion of making an internal `eagerConnectionShutdown` as `Boolean` and modifying the status only if it was explicitly set by users.